app/repositories/mails.py

```diff
@@ -0,0 +1,182 @@
+from __future__ import annotations
+
+import textwrap
+from typing import Any
+from typing import Optional
+
+import app.state.services
+
+# +--------------+------------------------+------+-----+---------+-------+
+# | Field        | Type                   | Null | Key | Default | Extra |
+# +--------------+------------------------+------+-----+---------+-------+
+# | id           | int                    | NO   | PRI | NULL    |       |
+# | from_id      | int                    | NO   |     | NULL    |       |
+# | to_id        | int                    | NO   |     | NULL    |       |
+# | msg          | varchar(2048)          | NO   |     | NULL    |       |
+# | time         | int                    | YES  |     | NULL    |       |
+# | read         | tinyint(1)             | NO   |     | NULL    |       |
+# +--------------+------------------------+------+-----+---------+-------+
+
+READ_PARAMS = textwrap.dedent(
+    """\
+        id, from_id, to_id, msg, time, `read`
+    """,
+)
+
+
+async def create(from_id: int, to_id: int, msg: str) -> dict[str, Any]:
+    """Create a new mail entry in the database."""
+    query = f"""\
+        INSERT INTO mail (from_id, to_id, msg, time)
+             VALUES (:from_id, :to_id, :msg, UNIX_TIMESTAMP())
+    """
+    params = {"from_id": from_id, "to_id": to_id, "msg": msg}
+    rec_id = await app.state.services.database.execute(query, params)
+
+    query = f"""\
+        SELECT {READ_PARAMS}
+          FROM mail
+         WHERE id = :id
+    """
+    params = {
+        "id": rec_id,
+    }
+    rec = await app.state.services.database.fetch_one(query, params)
+    assert rec is not None
+    return dict(rec)
```

Comment:
same feedback as https://github.com/osuAkatsuki/bancho.py/pull/489/files#r1335092434 for all the function returns in this pr